### PR TITLE
Enable bulk requests on row operations

### PR DESCRIPTION
**Pull request by urmilan**
Instead of executing delete operations by issuing a deleteByQuery on rowId (which isn't supported by `BulkRequest`), we look up the dataset column mappings and assemble a list of document IDs that need to be deleted. This enables us to do all row operations in bulk.

### CODE REVIEW

**Comment by urmilan** [comment]
Fixed merge conflicts with master and added another test.

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 1]
```diff
@@ -108,6 +108,15 @@ class SpandexElasticSearchClient(config: ElasticSearchConfig) extends ElasticSea
                            .setQuery(byDatasetIdQuery(datasetId))
                            .execute.actionGet)
 
+  // We don't expect the number of column maps to exceed config.dataCopyBatchSize.
+  // As of April 2015 the widest dataset is ~1000 cols wide.
```
Comment (velvia) [inline spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: Eeks, 1000 columns wide?

Comment (urmilan) [inline spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: Yes. According to @spaceballone this is an outlier and the next widest dataset is a couple of hundred columns wide.

**Comment by urmilan** [comment]
This is apparently the conflictiest PR. Merged in master again.

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/ResponseExtensions.scala [hunk 2]
```diff
@@ -80,7 +80,7 @@ case class FieldValue(datasetId: String,
                       columnId: Long,
                       rowId: Long,
                       value: String) {
-  lazy val docId = s"$datasetId|$copyNumber|$columnId|$rowId"
+  lazy val docId = FieldValue.makeDocId(datasetId, copyNumber, columnId, rowId)
   lazy val compositeId = s"$datasetId|$copyNumber|$columnId"
```
Comment (sokrahta) [inline spandex-common/src/main/scala/com/socrata/spandex/common/client/ResponseExtensions.scala]: in the same pattern: `FieldValue.makeCompositeId`

Comment (urmilan) [inline spandex-common/src/main/scala/com/socrata/spandex/common/client/ResponseExtensions.scala]: Done in 26c3475

**Comment by urmilan** [comment]
@velvia @sokrahta Are you guys ok for me to merge this?

**Comment by velvia** [comment]
LGTM, merging

**Pull request merged**